packages/cli-kit/src/public/node/json-schema.ts

```diff
@@ -31,27 +52,30 @@ const validatorsCache = new Map<string, ValidateFunction>()
  *
  * @param subject - The object to validate.
  * @param schema - The JSON schema to validate against.
+ * @param handleInvalidAdditionalProperties - Whether to strip or fail on invalid additional properties.
  * @param identifier - The identifier of the schema being validated, used to cache the validator.
  * @returns The result of the validation. If the state is 'error', the errors will be in a zod-like format.
  */
 export function jsonSchemaValidate(
   subject: object,
   schema: SchemaObject,
-  identifier: string,
+  handleInvalidAdditionalProperties: HandleInvalidAdditionalProperties,
+  identifier?: string,
 ): ParseConfigurationResult<unknown> & {rawErrors?: AjvError[]} {
-  const ajv = new Ajv({allowUnionTypes: true})
+  const subjectToModify = cloneDeep(subject)
 
-  ajv.addKeyword('x-taplo')
+  const cacheKey = identifier ?? randomUUID()
+  const ajv = createAjv(handleInvalidAdditionalProperties)
 
-  const validator = validatorsCache.get(identifier) ?? ajv.compile(schema)
-  validatorsCache.set(identifier, validator)
+  const validator = validatorsCache.get(cacheKey) ?? ajv.compile(schema)
```

Comment:
Is probably very fast, but we don't need to create a new ajv instance if we are going to use the cache right? Would it make sense to have a function called `createAjvValidator` that both creates the ajv instance and compiles the schema?